src/Opdex.Platform.Infrastructure/Data/Handlers/Markets/SelectMarketsWithFilterQueryHandler.cs

```diff
@@ -0,0 +1,226 @@
+                ? nameof(SortDirectionType.ASC)
+                : nameof(SortDirectionType.DESC),
+            _ => Enum.GetName(typeof(SortDirectionType), request.Cursor.SortDirection)
+        };
+
+        var orderBy = OrderByBuilder(request.Cursor.OrderBy, direction, false);
+
+        var limit = $" LIMIT {request.Cursor.Limit + 1}";
+
+        var query = SqlQuery
+            .Replace(PrimaryOrderColumn, primaryOrderColumn)
+            .Replace(WhereFilter, whereFilterBuilder.ToString())
+            .Replace(TableJoins, tableJoins)
+            .Replace(OrderBy, orderBy)
+            .Replace(Limit, limit).RemoveExcessWhitespace();
+
+        return request.Cursor.PagingDirection == PagingDirection.Forward
+            ? $"{query};"
+            : PagingBackwardQuery.Replace(InnerQuery, query)
+                                 .Replace(OrderBySort, OrderByBuilder(request.Cursor.OrderBy,
+                                     Enum.GetName(typeof(SortDirectionType), request.Cursor.SortDirection),
+                                     reverse: true));
+
+    }
+
+    private static string OrderByBuilder(MarketOrderByType cursorOrderBy, string direction, bool reverse)
+    {
+        var summaryPrefix = reverse ? "r" : "ms";
+        var marketPrefix = reverse ? "r" : "m";
+
+        var sortPart = cursorOrderBy switch
+        {
+            // we coalesce as market summary may be null, if the market has recently been created
+            MarketOrderByType.LiquidityUsd => $"{summaryPrefix}.{nameof(MarketSummaryEntity.LiquidityUsd)} {direction},",
+            MarketOrderByType.StakingUsd => $"{summaryPrefix}.{nameof(MarketSummaryEntity.StakingUsd)} {direction},",
+            MarketOrderByType.StakingWeight => $"{summaryPrefix}.{nameof(MarketSummaryEntity.StakingWeight)} {direction},",
+            MarketOrderByType.VolumeUsd => $"{summaryPrefix}.{nameof(MarketSummaryEntity.VolumeUsd)} {direction},",
+            MarketOrderByType.MarketRewardsDailyUsd => $"{summaryPrefix}.{nameof(MarketSummaryEntity.MarketRewardsDailyUsd)} {direction},",
+            MarketOrderByType.ProviderRewardsDailyUsd => $"{summaryPrefix}.{nameof(MarketSummaryEntity.ProviderRewardsDailyUsd)} {direction},",
+            MarketOrderByType.DailyLiquidityUsdChangePercent => $"{summaryPrefix}.{nameof(MarketSummaryEntity.DailyLiquidityUsdChangePercent)} {direction},",
+            MarketOrderByType.DailyStakingUsdChangePercent => $"{summaryPrefix}.{nameof(MarketSummaryEntity.DailyStakingUsdChangePercent)} {direction},",
+            MarketOrderByType.DailyStakingWeightChangePercent => $"{summaryPrefix}.{nameof(MarketSummaryEntity.DailyStakingWeightChangePercent)} {direction},",
+            _ => ""
+        };
+
+        var stringBuilder = new StringBuilder();
+        stringBuilder.Append(" ORDER BY ").Append(sortPart);
+        stringBuilder.Append(' ').Append(marketPrefix).Append('.').Append(nameof(MarketEntity.Id)).Append(' ').Append(direction);
```

Comment:
I think it would be easier on the eyes here to just return the string as `" ORDER BY ${sortPart} ${marketPrefix}.${nameof(MarketEntity.Id)} ${direction}"`

idk that we're getting too much benefit of a string builder here